src/IIIFPresentation/API/Features/Storage/StorageController.cs

```diff
@@ -20,39 +21,62 @@
 using Microsoft.AspNetCore.WebUtilities;
 using Microsoft.Extensions.Options;
 using Models.API.Collection;
-using Newtonsoft.Json;
+using Models.Database.General;
+using Repository;
+using Repository.Helpers;
 
 namespace API.Features.Storage;
 
 [Route("/{customerId:int}")]
 [ApiController]
-public class StorageController(IAuthenticator authenticator, IOptions<ApiSettings> options, IMediator mediator)
+public class StorageController(
+    IAuthenticator authenticator,
+    PresentationContext dbContext,
+    IOptions<ApiSettings> options,
+    IMediator mediator)
     : PresentationController(options.Value, mediator)
 {
-    private JsonSerializerSettings jsonSettings;
-    
     [HttpGet("{*slug}")]
     [ETagCaching]
     [VaryHeader]
-    public async Task<IActionResult> GetHierarchicalCollection(int customerId, string slug = "")
+    public async Task<IActionResult> GetHierarchical(int customerId, string slug = "")
     {
-        var storageRoot = await Mediator.Send(new GetHierarchicalCollection(customerId, slug, GetUrlRoots()));
+        var hierarchy =
```

Comment:
later please. It's (to me) a conundrum because you said "don't use mediator in mediator handler", here "do things in mediator handler", but tbh the only way left I saw when I wrote this was sth like `GetHierarchy : IRequest<Hierarchy?>` - the controller would dispatch this one, get result and then decide what comes next. But that seemed like 2 bonus classes to wrap one query.

I don't mind either way, but I ran out of ideas :D